src/manufacturer/manufacturerDialog.component.tsx

```diff
@@ -160,19 +205,158 @@ function AddManufacturerDialog(props: AddManufacturerDialogProps) {
           setNameErrorMessage(
             'A manufacturer with the same name already exists.'
           );
+          return;
         }
+        setCatchAllError(true);
       });
-  }, [manufacturer, addManufacturer, handleClose]);
+  }, [handleErrors, manufacturer, addManufacturer, handleClose]);
+
+  const handleEditManufacturer = React.useCallback(() => {
+    if (selectedManufacturer && selectedManufacturerData) {
+      const hasErrors = handleErrors();
+
+      if (hasErrors) {
+        return;
+      }
+
+      const isNameUpdated = manufacturer.name !== selectedManufacturer.name;
+      const isURLUpdated =
+        manufacturer.url !== selectedManufacturer.url &&
+        manufacturer.url !== undefined;
+      const isBuildingNumberUpdated =
+        manufacturer.address?.building_number !==
+        selectedManufacturer.address.building_number;
+      const isStreetNameUpdated =
+        manufacturer.address?.street_name !==
+        selectedManufacturer.address.street_name;
+      const isTownUpdated =
+        manufacturer.address?.town !== selectedManufacturer.address.town;
+      const isCountyUpdated =
+        manufacturer.address?.county !== selectedManufacturer.address.county;
+      const ispostcodeUpdated =
+        manufacturer.address?.postcode !==
+        selectedManufacturer.address.postcode;
+      const isTelephoneUpdated =
+        manufacturer.telephone !== selectedManufacturer.telephone;
+
+      let ManufacturerToEdit: EditManufacturer = {
+        id: selectedManufacturer?.id,
+      };
+
+      if (isNameUpdated) {
+        ManufacturerToEdit = {
+          ...ManufacturerToEdit,
+          name: manufacturer.name,
+        };
+      }
```

Comment:
Just spoken to Joshua, these could also be simplified slightly
```suggestion
     isNameUpdated && (ManufacturerToEdit .name = manufacturer.name)
```
But it wont work for the address as its nested. Best to address this on  #118 though.

Reply:
Addressed in #118 with commit "simplified edit #73"